test/hotspot/jtreg/runtime/cds/appcds/SharedRegionAlignmentTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @requires vm.cds
+ * @requires vm.gc != "Z"
+ * @summary Testing handling of CDS region alignment
```

Comment:
I think the summary should be more specific, such as "Make sure CDS core region alignment is at least 64KB".

Also, I think we should add

```
* @comment ZGC may exit the VM if -XX:+UseLargePages is specified but
*          unavailable. Since this test is independent of the actual GC type, let's
*          disable it if ZGC is used.
```